src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -312,16 +313,20 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     } getOrElse Future.successful(())
   }
 
-  private[service] def validateBucketObjectUri(googleProject: GoogleProject, gcsUriOpt: Option[GcsPath])(implicit executionContext: ExecutionContext): Future[Unit] = {
+  private[service] def validateBucketObjectUri(userEmail: WorkbenchEmail, googleProject: GoogleProject, gcsUriOpt: Option[GcsPath])(implicit executionContext: ExecutionContext): Future[Unit] = {
+
     gcsUriOpt match {
       case None => Future.successful(())
       case Some(gcsPath) =>
         if (gcsPath.toUri.length > bucketPathMaxLength) {
           throw BucketObjectException(gcsPath)
         }
-        googleStorageDAO.objectExists(gcsPath.bucketName, gcsPath.objectName).map {
-          case true => ()
-          case false => throw BucketObjectException(gcsPath)
+        serviceAccountProvider.getAccessToken(userEmail, googleProject).flatMap {
+          case Some(token) => petGoogleStorageDAO(token).objectExists(gcsPath.bucketName, gcsPath.objectName).map {
```

Comment:
@vkumra-broad could we add a recover to `objectExists` and provide a better error message in case of a 403 from Google? @calbach hit this bug recently and got a generic 500 from Leo with no additional information.